Sources/Operators/RemoveKnownDuplicates.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  RemoveKnownDuplicates.swift
+//  CombineExt
+//
+//  Created by Jasdev Singh on 21/03/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+public extension Publisher where Output: Hashable {
+    /// When `Output` conforms to `Hashable`, `Publisher.removeKnownDuplicates` de-duplicates _all_
+    /// published value events, as opposed to pairwise with `Publisher.removeDuplicates`.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Set`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events.
+    ///
+    /// - returns: A type-erased publisher that consumes duplicate values across all previous emissions from upstream.
+    func removeKnownDuplicates() -> AnyPublisher<Output, Failure> {
+        scan((Set<Output>(), Output?.none)) { seenAndPrevious, incoming in
+            var seen = seenAndPrevious.0
+            let isUnique = seen.insert(incoming).inserted
+            return (seen, isUnique ? incoming : nil)
+        }
+        .compactMap { $0.1 }
+        .eraseToAnyPublisher()
+    }
+}
+
+public extension Publisher where Output: Equatable {
+    /// When `Output` conforms to `Equatable`, `Publisher.removeKnownDuplicates` de-duplicates _all_
+    /// published value events, as opposed to pairwise with `Publisher.removeDuplicates`.
```

Comment:
Same re-docs. If we're noting specifically re-Hashable/Equatable, it's better to put it at the end of the sentence, like 

(just with better line break lol)
```suggestion
    /// `Publisher.removeKnownDuplicates` de-duplicates _all_
    /// published `Hashable`-conforming value events, as opposed to pairwise with `Publisher.removeDuplicates`.
```